archive/tar_opts.go

```diff
@@ -73,3 +73,13 @@ func WithParents(p []string) ApplyOpt {
 		return nil
 	}
 }
+
+// WriteDiffOptions provides additional options for a WriteDiff operation
+type WriteDiffOptions struct {
+	ParentLayers []string // Windows needs the full list of parent layers
```

Comment:
Should this be `WriteDiffOptions` in `tar_opts_windows.go` since its Windows specific?

Reply:
The structure needs to be visible in the cross-platform code as it's passed to `archive.WriteDiff`, unless we want to instead complete decouple WCOW Layer Diff/Apply from using `archive` APIs. (Which I'd be fine with, but is a much wider change, and affects existing APIs)

The particular field is only used on Windows, and I'm not sure it's more valuable than confusing to bifurcate the structure by defining `WriteDiffOptions` differently for `windows` and `!windows`. The same argument could be made for `applyFunc` and `writeDiffFunc`, so I'd rather punt this to later possible refactoring as those are much-worse warts, and there are much greater improvements available than splitting this structure.

An alternative approach to removing this field would be to smuggle this array of strings via the `a` (or `lower`) string parameter which is currently unused in this flow. I think I saw code somewhere around here which was doing that by serialising the array into JSON and the unserialising it once back in platform-specific code.

I was not fond of that...

Reply:
Np